zipkin-lens/src/components/DiscoverPage/LookbackMenu.tsx

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2015-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import {
+  Box,
+  Button,
+  Grid,
+  List,
+  ListItem,
+  ListItemText,
+  Paper,
+  Theme,
+  createStyles,
+  makeStyles,
+} from '@material-ui/core';
+import { KeyboardDateTimePicker } from '@material-ui/pickers';
+import { MaterialUiPickersDate } from '@material-ui/pickers/typings/date';
+import moment, { Moment } from 'moment';
+import React, { useCallback, useRef, useState } from 'react';
+import { useEvent } from 'react-use';
+
+import { fixedLookbackMap, FixedLookbackValue, Lookback } from './lookback';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      position: 'absolute',
+      top: 35,
+      left: 0,
+      height: 300,
+      width: 500,
+      zIndex: theme.zIndex.modal,
+    },
+    containerGrid: {
+      height: '100%',
+    },
+    fixedLookbackItemGrid: {
+      height: '100%',
+      overflowY: 'auto',
+      borderRight: `1px solid ${theme.palette.divider}`,
+    },
+    list: {
+      padding: 0,
+    },
+  }),
+);
+
+interface LookbackMenuProps {
+  close: () => void;
+  onChange: (lookback: Lookback) => void;
+  lookback: Lookback;
+}
+
+const initialStartTime = (lookback: Lookback): Moment => {
+  if (lookback.type === 'custom') {
+    return lookback.startTime;
+  }
+  return moment().subtract(1, 'h');
+};
+
+const initialEndTime = (lookback: Lookback): Moment => {
+  if (lookback.type === 'custom') {
+    return lookback.endTime;
+  }
+  return moment();
+};
+
+const LookbackMenu: React.FC<LookbackMenuProps> = ({
+  close,
+  onChange,
+  lookback,
+}) => {
+  const classes = useStyles();
+
+  // LookbackMenu is closed when click on the outside of the component.
+  // This state is needed to prevent LookbackMenu component from closing
+  // when the dialog of DateTimePicker is clicked.
+  const [isOpeningDialog, setIsOpeningDialog] = useState(false);
+
+  const handleDialogOpen = useCallback(() => {
+    setIsOpeningDialog(true);
+  }, []);
+
+  const handleDialogClose = useCallback(() => {
+    // Use setTimeout to change isOpeningDialog state after
+    // handleOutsideClick callback function is executed.
```

Comment:
Why is this important?

Reply:
If there is no setTimeout, LookbackMenu will also be closed when the DateTimePicker's dialog is clicked.

![without-settimeout](https://user-images.githubusercontent.com/19551419/87141773-89e16600-c2de-11ea-8728-096b64d76ac0.gif)

I use setTimeout to achieve the following behavior

![with-settimeout](https://user-images.githubusercontent.com/19551419/87141863-b0070600-c2de-11ea-92f8-5a5653b5e9bd.gif)